### PR TITLE
Upgrading to Yarn 4.1.1

### PR DESCRIPTION
**Pull request by morganick**
## Please verify the following:

- [x] `yarn build-and-test:local` passes
- [ ] ~I have added tests for any new features, if relevant~
- [x] `README.md` (or relevant documentation) has been updated with your changes

## Describe your PR
This PR moves us to the latest stable version of Yarn version 4.1.1.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 nodeLinker: node-modules
 # allow NPM_TOKEN env to be set, but provide '' as fallback. This allows CI to publish packages, but not require env var.
 # See https://yarnpkg.com/configuration/yarnrc
 npmAuthToken: "${NPM_TOKEN-}"
 
 npmPublishAccess: public
 
-plugins:
-  - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
-    spec: "@yarnpkg/plugin-interactive-tools"
```
Comment (morganick) [inline .yarnrc.yml:10]: This plugin is automatically included in Yarn 4. See: https://www.npmjs.com/package/@yarnpkg/plugin-interactive-tools#install

File: .yarnrc.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
+compressionLevel: mixed
```
Comment (morganick) [inline .yarnrc.yml:1]: https://yarnpkg.com/configuration/yarnrc#compressionLevel

File: apps/example-app/package.json [hunk 3]
```diff
@@ -76,6 +76,7 @@
     "@babel/runtime": "^7.20.0",
     "@types/i18n-js": "3.8.2",
     "@types/jest": "^29.2.1",
+    "@types/node": "^20.11.30",
```
Comment (morganick) [inline apps/example-app/package.json:79]: `ts-node` was complaining about this not being installed on `yarn install`.

File: package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
   },
   "resolutions": {
     "@nx/devkit": "17.0.3",
-    "@types/react": "18.2.35",
+    "@types/react": "18.2.45",
```
Comment (morganick) [inline package.json:51]: This was causing a warning inside the example app so I've upgraded this across the board.

**Comment by morganick** [comment]
@markrickert and @joshuayoes any feels on this?

File: .yarnrc.yml (hunk 1 above)
Comment (markrickert) [reply .yarnrc.yml:10]: Sweet! I think that's for things like `yarn upgrade-interactive`

**Review by markrickert — APPROVED** [review]
Nice! :rubber-stamp-emoji:

I pulled down the branch and ran the clean & build and everything looks good! Example app and electron apps run as expected with the yarn upgrade.

**Pull request merged**